pandas/core/internals/managers.py

```diff
@@ -1500,6 +1511,8 @@ def __init__(
         do_integrity_check: bool = False,
         fastpath: bool = False,
     ):
+        assert isinstance(block, Block), type(block)
```

Comment:
@simonjayhawkins is mypy reliable enough that we can remove this assertion?  these add up in things like groupby.apply when we create many many Series objects

Reply:
> is mypy reliable enough that we can remove this assertion?

not really as we don't have very strict type checking.

we could call the constructor with a variable that has type Any, or we could call the constructor from a function that is not currently type checked (still don't have check_untyped_defs for many modules) 

to prevent calling the constructor with a variable with type Any, we would need disallow dynamic typing completely, https://mypy.readthedocs.io/en/stable/config_file.html#disallow-dynamic-typing